contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/genio_performance_mode.py

```diff
@@ -0,0 +1,436 @@
+    ):
+        node_path = PATH_OF_THERMAL_TRIP_POINT.format(
+            thermal_zone, trip_point_count
+        )
+        self._set_node_value(node_path=node_path, value=temp_value)
+
+    def _set_apusys(self, value):
+        node_path = "/sys/kernel/debug/apusys/power"
+        self._set_node_value(node_path=node_path, value=value)
+
+    def _toggle_cpuidle_state_disable(self, cpu_count, state_count, value):
+        """
+        Toggle the 'disable' attribute for CPU idle states.
+
+        Args:
+            cpu_count (int): The count of CPUs.
+            state_count (int): The count of CPU idle stat.
+            value (int):
+                The value to set for the 'disable' attribute (0 or 1).
+
+        Raises:
+            ValueError: If the provided value is not 0 or 1.
+        """
+        if value not in [0, 1]:
+            raise ValueError("Value must be 0 or 1")
+        self._set_node_value(
+            PATH_CPUIDLE_STATE_DISABLE.format(cpu_count, state_count), value
+        )
+
+    def _toggle_thermal_mode(self, thermal_zone, value):
+        if value not in ["disabled", "enabled"]:
+            raise ValueError("Thermal mode must be enabled or disabled")
+        self._set_node_value(PATH_OF_THERMAL_MODE.format(thermal_zone), value)
+
+
+class PerformanceModeManager(PerformanceController):
+    """
+    PerformanceModeManager class for managing system performance modes.
+
+    This class extends the PerformanceController class and provides methods
+    to set and restore performance modes for CPU, GPU, APU, and thermal
+    configurations.
+
+    Args:
+        affected_policies (list): List of affected CPU policies.
+        affected_thermal_zone (int): Thermal zone to be affected.
+        affected_thermal_trip_points (list):
+            List of thermal trip points to be affected.
+        affected_trip_point_temp (int):
+            Temperature value for affected thermal trip points.
+        change_thermal_mode (bool):
+            Flag to indicate if thermal mode should be changed.
+        gpu_soc_name (str): Name of the GPU SoC to be affected.
+        enable_apu (bool): Flag to indicate if APU should be enabled.
+
+    Methods:
+        set_performance_mode():
+            Configures CPU, GPU, APU and thermal settings for performance mode.
+
+        restore_default_mode():
+            Restores default CPU, GPU and thermal settings.
+
+    Note:
+        This class assumes the existence of methods such as
+        _backup_cpu_governor, _set_cpu_governor, _toggle_cpuidle_state_disable,
+        _backup_gpu_governor, _set_gpu_governor, _set_apusys,
+        _toggle_thermal_mode, _backup_thermal_trip_point_temp,
+        _set_thermal_trip_point_temp, _restore_cpu_governor,
+        _restore_gpu_governor, and _restore_thermal_trip_point_temp
+        in the parent class (PerformanceController).
+    """
+
+    def __init__(
+        self,
+        affected_policies: list,
```

Comment:
The raw `list` annotation is not valid syntax until python3.9. We would need either:
```py
from typing import List

def __init__(
    affected_policies: List
)
``` 
or
```py
def __init__(
    affected_policies: 'list'
)
```
If we are using >3.9 this is not an issue :D

Reply:
This script will be removed because it was implemented in Baoshan Project. I would fix this problem in it. Thanks